java-checks/src/main/java/org/sonar/java/checks/DepthOfInheritanceTreeCheck.java

```diff
@@ -71,9 +77,11 @@ public void visitClass(ClassTree tree) {
     if (!isBodyOfEnumConstantTree(tree)) {
       Type superClass = tree.symbol().superClass();
       int dit = 0;
+      List<WildcardPattern> exclusionPatterns = getPatterns();
```

Comment:
This is not equivalent to what we were doing with `getPatterns()` before. We need to cache the patterns in the instance of the check to not recompute them every time we enter a class.